### PR TITLE
support for affine conic constraints (Ax+b in K)

### PR DESCRIPTION
**Pull request by ulfworsoe**
@blegat This is my suggestion. I have not tested *at all* so it is almost certain to not work yet, and I probably forgot some things.

One note: 
- Currently it is not possible to remove AffineConicConstraint entries in MOSEK. It *is* possible to reuse then and reset the domain, but in this patch I only *clear* the ACCs in `MOI.delete()`, I never reuse them.
- Domains in MOSEK are immutable and they cannot be deleted

### CODE REVIEW

File: src/MosekTools.jl [hunk 1]
```diff
@@ -46,6 +49,32 @@ struct MatrixIndex
     end
 end
 
+type ACCVectorDomain = Union{MOI.Reals,
```
Comment (blegat) [inline src/MosekTools.jl:52]: `type` -> `const`

File: src/attributes.jl [hunk 2]
```diff
@@ -426,6 +425,57 @@ function MOI.get(m     ::MosekModel,
...
+end
+function permute_sol_mosek_to_julia(::Type{D},vals :: Vector{Float64})
+    where {D<:Union{MOI.ExponentialCone,
+                    MOI.DualExponentialCone}}
+    return Float64[vals[3],vals[2],vals[1]]
+end
+function permute_sol_mosek_to_julia(::Type{MOI.PositiveSemidefiniteConeTriangle},vals :: Vector{Float64})
+    N = length(vals)
+    res = Vector{Float64}(N)
+    d = Int(floor((-1.0+sqrt(1.0+8.0*N))/2.0))
```
Comment (blegat) [inline src/attributes.jl:451]: You can use `d = MOI.Utilities.side_dimension_for_vectorized_dimension(N)`

**Comment by ulfworsoe** [comment]
I would like to ask about the top-level `if Mosek.getversion() >= 10`: Do they get evaluated only when the module is loaded from source? And would an update of Mosek cause it to be reevaluated?

File: src/constraint.jl [hunk 3]
```diff
@@ -430,6 +444,147 @@ function MOI.add_constraint(m::MosekModel, xs::MOI.VectorOfVariables,
...
+    axb = MOIU.canonical(axb)
+    set_afe_rows(m,afeidxs,axb)
+
+    accid = Mosek.getnumacc(m.task)+1
+    Mosek.appendacc(m.task,
+                    domidx,
+                    index_julia_to_mosek(afeidxs,dom),
+                    zeros(Float64,N))
+
+    ci = MOI.ConstraintIndex{MOI.VectorAffineFunction,D}(accid)
```
Comment (blegat) [inline src/constraint.jl:583]: Add `{Float64}` after `MOI.VectorAffiveFunction`

Comment (ulfworsoe) [reply src/constraint.jl:583]: Obvious when you point it out. Thanks!

**Pull request closed without merging**